model/src/main/kotlin/ScanResults.kt

```diff
@@ -0,0 +1,151 @@
+ */
+data class ScanResults(
+        /**
+         * The [Identifier] of the package these [results] belong to.
+         */
+        val id: Identifier,
+
+        /**
+         * The list of [ScanResult]s.
+         */
+        val results: List<ScanResult>
+)
+
+/**
+ * The result of a single scan of a package.
+ */
+data class ScanResult(
+        /**
+         * Provenance information about the scanned source code.
+         */
+        val provenance: Provenance,
+
+        /**
+         * Details about the used scanner.
+         */
+        val scanner: ScannerSpecification,
+
+        /**
+         * A summary of the scan results.
+         */
+        val summary: ScanSummary,
+
+        /**
+         * The raw output of the scanner.
+         */
+        val rawResult: JsonNode
+)
+
+/**
+ * Provenance information about the scanned source code.
+ */
+data class Provenance(
+        /**
+         * The time when the source code was downloaded.
+         */
+        val downloadTime: Instant,
+
+        /**
+         * The source artifact that was downloaded, or null.
+         */
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        val sourceArtifact: RemoteArtifact? = null,
+
+        /**
+         * The VCS repository that was downloaded, or null.
+         */
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        val vcsInfo: VcsInfo? = null,
+
+        /**
+         * True if a local directory with no provenance information was scanned. Only serialized if the value is true.
+         */
+        @JsonInclude(JsonInclude.Include.CUSTOM, valueFilter = FalseFilter::class)
+        val localScan: Boolean = false
+)
+
+/**
+ * Details about the used source code scanner.
+ */
+data class ScannerSpecification(
+        /**
+         * The name of the scanner.
+         */
+        val name: String,
+
+        /**
+         * The version of the scanner.
+         */
+        val version: String,
+
+        /**
+         * The configuration of the scanner, could be command line arguments for example.
+         */
+        val configuration: String
+)
+
+/**
+ * A short summary of the scan results.
+ */
+data class ScanSummary(
+        /**
+         * The time when the scan started.
+         */
+        val startTime: Instant,
+
+        /**
+         * The time when the scan finished.
+         */
+        val endTime: Instant,
+
+        /**
+         * The number of scanned files.
+         */
+        val fileCount: Int,
+
+        /**
+         * A list of licenses detected by the scanner.
+         */
+        val licenses: SortedSet<String>,
+
+        /**
+         * A list of errors that occured during the scan.
+         */
+        val errors: SortedSet<String>
+)
+
+/**
+ * A filter to be used with [JsonInclude.Include.CUSTOM] to not serialize boolean properties set to false.
+ */
+private class FalseFilter {
+    override fun equals(other: Any?) = other != null && other is Boolean && !other
```

Comment:
You can omit the `null` check as you already check for the (non-nullable) `Boolean` type.

Reply:
Removed the whole class instead because it is not needed anymore :)